### DISCUSSION TITLE
Is there possible open helm selected file in a new window horizontally?

### DISCUSSION THREAD

**Discussion by zw963** [emacs-helm/helm#2555] [Q&A] [answered]
Current solution, e.g. see following screenshot.

![image](https://user-images.githubusercontent.com/549126/195346733-ce9f6de0-3e8d-4080-b576-33ae27280928.png)

When i pressing `C-c o`, it open the new buffer window vertically, like this.

![image](https://user-images.githubusercontent.com/549126/195346917-8911ea42-f1e2-4adb-853d-bc3630fbb130.png)

is there possible always open new file buffer window right?

Thanks you.

**Comment by thierryvolpiatto** [comment]
"Billy.Zheng" ***@***.***> writes:

> is there possible always open new file buffer window right?

C-u C-c o

-- 
Thierry

**Reply by zw963** [reply]
Yes, It works, thank you, can i set it as default?

**Comment by thierryvolpiatto** [comment]
![Capture d’écran_2022-10-12_17-30-43](https://user-images.githubusercontent.com/1533939/195385753-5962566a-4e7a-4ae5-a98d-447601588792.png)

**Comment by thierryvolpiatto** [comment, marked as answer]
"Billy.Zheng" ***@***.***> writes:

> Yes, It works, thank you, can i set it as default?

See `helm-window-prefer-horizontal-split`

And then prefix arg behavior will be inversed.

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Reply by zw963** [reply]
Cool, i know this option, but i thought it will make helm window open right too, that not good.

But, when i try it, helm still bottom, but open file buffer right, work like a charm!

i guess this probably relative to following config:

```
;; 让 helm 的弹出窗口有类似于 popwin 的效果。
(setq helm-display-function 'pop-to-buffer) ; make helm play nice
(add-to-list 'shackle-rules '("\\`\\*helm.*?\\*\\'" :regexp t :align t :size 0.618))
```

**Comment by thierryvolpiatto** [comment]
"Billy.Zheng" ***@***.***> writes:

> Cool, i know this option, but i thought it will make helm window open right too, that not good.
>
> But, when i try it, helm still bottom, but open file buffer right, work like a charm!
>
> i guess this probably relative to following config:
>
> ;; 让 helm 的弹出窗口有类似于 popwin 的效果。
> (setq helm-display-function 'pop-to-buffer) ; make helm play nice

Helm assume you are using either `helm-default-display-buffer` or
`helm-display-buffer-in-own-frame`, if you use `pop-to-buffer`, you will
have to implement all yourself with display-buffer-alist, good luck ;-).
See `helm-default-display-buffer-alist`.
Also using pop-to-buffer will prevent many cool helm stuff from working
properly e.g. C-t, `helm-commands-using-frame` etc...

> (add-to-list 'shackle-rules '("\\`\\*helm.*?\\*\\'" :regexp t :align t
> :size 0.618))

I don't know this so I can't tell.

-- 
Thierry

**Reply by zw963** [reply]
> I don't know this so I can't tell.

anyway, my current solution work very well for me usage, thank you very much!
